lib/domain/models/resource.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:memo/data/database_repository.dart';
+import 'package:meta/meta.dart';
+
+@immutable
+class Resource extends KeyStorable {
+  Resource({required String id, required this.description, required this.tags, required this.url})
+      : assert(tags.isNotEmpty, 'tags must have at least one element'),
```

Comment:
I think that today we don't have any case of a Resource with empty URL, so we could assert it here as well

Reply:
Also, we must differ the resources types so we can customize them in the UI like `video`, `article`, `book`, etc..

Reply:
I don't think these assertions should be this fine-grained, because this opens to discussion: well, it should also be a valid URL, and it should always be HTTPS, and the list goes on. So I don't think we should validate strings at all, other than they should be null-safe required.